mongo/bulk_write.go

```diff
@@ -248,7 +248,7 @@ func (bw *bulkWrite) runDelete(ctx context.Context, batch bulkWriteBatch) (opera
 func createDeleteDoc(filter interface{}, collation *options.Collation, hint interface{}, deleteOne bool,
 	registry *bsoncodec.Registry) (bsoncore.Document, error) {
 
-	f, err := transformBsoncoreDocument(registry, filter)
+	f, err := transformBsoncoreDocument(registry, filter, true)
```

Comment:
Commenting here so we don't lose track, but can we add a short note to all functions where an unordered map is prohibited? `DeleteOneModel.SetHint` does not have any such documentation. It could be as simple as "Passing an unordered map is prohibited" or something like that.